File: haskell/1.hs
-- PROBLEM 1:
--
-- If we list all the natural numbers below 10 that are multiples of 3 or 5, we
-- get 3, 5, 6 and 9. The sum of these multiples is 23.
--
-- Find the sum of all the multiples of 3 or 5 below 1000.

isvalid :: Int -> Bool
isvalid x | mod x 3 == 0  = True
          | mod x 5 == 0  = True
          | otherwise     = False

main = do
    print $ sum ( filter isvalid [1..999] )

File: go/1.go
package main

import "fmt"

const maxint = 1000
var sum int

func main() {
    for i := 1 ; i < maxint ; i++ {
        if i % 3 == 0 || i % 5 == 0 {
            sum += i
        }
    }
    fmt.Println(sum)
}
